### ISSUE TITLE
Long live Rethinkdb: Rethinkdb on the client as a great alternative to Horizon

### ISSUE THREAD

**Issue by babakness** [rethinkdb/rethinkdb#6138]
While Horizon is a offshoot project (really cool one at that), this project strives to bring RethinkDB / ReQL directly into the client over websockets

https://github.com/mikemintz/rethinkdb-websocket-client

Works pretty good. We've had success with it in React Native and it works for us with Redux, RxJS and/or Redux Observables (https://github.com/redux-observable/redux-observable)

I would bet that it will work just fine in NativeScript and Electron as well.

Anyway, I hope other people in community can consider this too, I wonder if many don't know about it--it is much closer to the origins of the project than Horizon is. You get ReQL directly in the browser. On the server / security side of things, you have this project

https://github.com/mikemintz/rethinkdb-websocket-server

This translates the TCP connection to websocket; It uses a whitelist to allow only specific queries through. Its a clever setup and could easily be turned into a "serverless" setup where the user just submits the whitelist functions and adds hooks to before and after the data is processed as well as on error triggers.

Here is an example of it in use through another project by Mike that wraps the two aforementioned projects into a React Mixin for rapid prototyping:

https://www.youtube.com/watch?v=WchhRxdNojk

**Comment by geekyme** [comment]
Please use GitHub issues only to file bugs related to rethinkdb, not market other repos. For those purposes, try posting in the slack group. 

http://slack.rethinkdb.com/

**Comment by babakness** [comment]
@geekyme Well, I disagree with the allegation of marketing, its not my repo anyway. Arguably Horizon is a different repo too yet maintained by the original RethinkDB team. This effort here is basically an extension of what RethinkDB can do. 

Increasing interest in RethinkDB is in the interest of all who want to see it move forward.

For sake of strategy, consider that Meteor drives a lot of commercial business to MongoDB. Yet its basically an elaborate hack. RethinkDB offers an elegant solution for the community in this regard.  Where MiniMongo fails to effectively work in React Native, this tool works well. Its also far more powerful than MiniMongo. Writing ReQL queries directly from a native app is pretty powerful. 

TLDR; this isn't about marketing. Its about strategy and increasing interest. Writing ReQL directly in the client--React Native, Electron, etc., is powerful and unmatched.

**Comment by kofalt** [comment]
As someone who isn't in the slack group & wouldn't have time to keep track of a real-time feed, thank you for filing this announcement ticket! I'd much rather receive one more github notification than one fewer tool in my collection :)

**Comment by kamilion** [comment]
Ooh, perfect, just what I was looking for. Also, slack is terrible, rethinkdb doesn't pay for it so the message history only goes back 10000 entries. Most of that history is now lost, nobody's archived it, there's no logs that I can find, unlike the IRC channel, which has been dead as a doorknob since the whole "support" thing was introduced.
